pyfarm/jobtypes/core/internals.py

```diff
@@ -455,7 +455,15 @@ def _process_stopped(self, protocol, reason):
         logger.info("%r stopped (code: %r)", protocol, reason.value.exitCode)
         process_data = self.processes.pop(protocol.uuid)
 
-        if self.is_successful(reason):
+        try:
+            successful = self.is_successful(protocol, reason)
+        except Exception as e:
+            logger.error("Exception caught from is_successful: %s: \"%s\". "
```

Comment:
You could construct the error message once and use it in both places.  I'd also use `%r` instead of manually escaping quotes.